libraries/AP_Mount/AP_Mount_XFRobot.cpp

```diff
@@ -222,6 +222,27 @@ bool AP_Mount_XFRobot::take_picture()
     return send_simple_command(FunctionOrder::SHUTTER, 0x01);
 }
 
+//set camera lens
+bool AP_Mount_XFRobot::set_lens(uint8_t lens)
+{
+    // lens to camera type mapping table
+    static const CameraType cam_type_table[] {
+        CameraType::MAIN_ZOOM_SUB_THERMAL,
+        CameraType::MAIN_THERMAL_SUB_ZOOM,
+        CameraType::MAIN_PIP_ZOOM_SUB_THERMAL,
+    };
+
+    // sanity check lens values
+    if (lens >= ARRAY_SIZE(cam_type_table)) {
+        return false;
+    }
+
+    // map lens to camera type
+    const CameraType selected_cam = cam_type_table[lens];
+
+    return send_simple_command(FunctionOrder::PIC_IN_PIC, (uint8_t)selected_cam);
```

Comment:
You're not taking a reference to the thing in the table here.  That's not really a problem as they're currently just enumeration values.  But you could just fold these lines together and avoid a potential future bug too.

Reply:
cool, txs for the review, I've combined the lines together and re-pushed over @Ferruccio1984's branch, thanks!